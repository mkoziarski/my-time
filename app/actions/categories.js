// @flow
import type { EditCategoryAction } from '../reducers/types';

export const EDIT_CATEGORY = 'EDIT_CATEGORY';

export function editCategory(id: string, name: string): EditCategoryAction {
  return {
    type: EDIT_CATEGORY,
    id: id,
    name: name
  }
}
