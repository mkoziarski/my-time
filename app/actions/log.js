// @flow
import type { StartLogEntryAction, GetState, Dispatch } from '../reducers/types';

export const START_LOG_ENTRY = 'START_LOG_ENTRY';

export function startLogEntry(categoryId: string): StartLogEntryAction {
  return {
    type: START_LOG_ENTRY,
    categoryId: categoryId
  }
}
