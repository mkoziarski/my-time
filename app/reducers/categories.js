// @flow
import uuid from 'uuid/v4';
import type { Categories, Action } from './types';
import { EDIT_CATEGORY } from '../actions/categories';

const initialCategories: Categories = [
  { id: uuid(), name: 'hoovering'},
  { id: uuid(), name: 'washing up'},
  { id: uuid(), name: 'laundry'},
  { id: uuid(), name: 'shopping'},
  { id: uuid(), name: 'cooking'},
  { id: uuid(), name: 'meals'},
  { id: uuid(), name: 'time with Edzilla'}
];

export default function categories(state: Categories = initialCategories, action: Action): Categories {
  switch (action.type) {
    case EDIT_CATEGORY:
      const id = action.id;
      const name = action.name;

      return state.map((category) => {
        // Flow fails if we use action.id and action.name in this context
        // doesn't recognize that action is an EditCategoryAction
        // we use bindings created in the parent scope instead
        if (category.id === id) {
          return {
            ...category,
            name: name
          };
        }

        return category;
      });
    default:
      return state;
  }
};
