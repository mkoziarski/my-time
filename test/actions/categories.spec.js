import * as categoryActions from '../../app/actions/categories';

describe('category actions', () => {
  test('editCategory returns an EDIT_CATEGORY action', () => {
    const id = '1a1eb070-d55a-4bb9-a094-6529633e0fa0';
    const name = 'categoryName';

    expect(categoryActions.editCategory(id, name)).toMatchSnapshot();
  });
});
