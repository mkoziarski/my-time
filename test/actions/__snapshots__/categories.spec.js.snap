// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category actions editCategory returns an EDIT_CATEGORY action 1`] = `
Object {
  "id": "1a1eb070-d55a-4bb9-a094-6529633e0fa0",
  "name": "categoryName",
  "type": "EDIT_CATEGORY",
}
`;
